docs/user-docs/build_linux.md

```diff
@@ -0,0 +1,172 @@
+echo '# CONFIG_RANDOMIZE_BASE is not set' >> "$KBUILD_OUTPUT/.config"
+```
+
+Select strict IOMMU translated mode by default for better device isolation at
+the cost of performance (can be omitted or enabled via `iommu.strict=1` kernel
+parameter):
+
+```bash
+echo CONFIG_IOMMU_DEFAULT_DMA_STRICT=y >> "$KBUILD_OUTPUT/.config"
+```
+
+Enable at least one driver for a TPM without which no DRTM can be done:
+
+```bash
+# this might be enough (e.g., for firmware TPM in recent CPUs)
+echo CONFIG_TCG_TPM=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_CRB=y >> "$KBUILD_OUTPUT/.config"
+
+# this enables the rest of TPM drivers
+echo CONFIG_TCG_TIS=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_TIS_I2C=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_TIS_I2C_CR50=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_TIS_I2C_ATMEL=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_TIS_I2C_INFINEON=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_TIS_I2C_NUVOTON=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_NSC=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_ATMEL=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_INFINEON=y >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_TCG_TIS_ST33ZP24_I2C=y >> "$KBUILD_OUTPUT/.config"
+```
+
+TPM drivers are located in
+`Device Drivers → Character devices → TPM Hardware Support` menu where more
+detailed information about the above options can be found.
+
+Enable Secure Launch itself:
+
+```bash
+echo CONFIG_SECURE_LAUNCH=y >> "$KBUILD_OUTPUT/.config"
+```
+
+Add a suffix to the version to avoid conflicts, also disable appending of commit
+hash:
+
+```bash
+echo 'CONFIG_LOCALVERSION="-v11-tb"' >> "$KBUILD_OUTPUT/.config"
+echo CONFIG_LOCALVERSION_AUTO=n >> "$KBUILD_OUTPUT/.config"
+```
+
+Now make the configuration usable by the build system by completing it:
+
+```bash
+make olddefconfig
+```
+
+[secure_launch_details]: https://github.com/TrenchBoot/linux/blob/linux-sl-master-9-12-24-v11/Documentation/security/launch-integrity/secure_launch_details.rst
+
+## Build
+
+```bash
+# the kernel
+make -j$(nproc) bzImage
+
+# and its modules
+make -j$(nproc) modules
+```
+
+## Installation
+
+The following steps should be run outside of a container either as root user or
+with `sudo` prepended to them:
+
+```bash
+# kernel
+cp "$KBUILD_OUTPUT/arch/x86/boot/bzImage" /boot/vmlinuz-6.11-rc7-v11-tb
+cp "$KBUILD_OUTPUT/.config" /boot/config-6.11-rc7-v11-tb
+cp "$KBUILD_OUTPUT/System.map" /boot/System.map-6.11-rc7-v11-tb
+
+# modules
+make modules_install
+```
+
+## Use with TrenchBoot GRUB2
+
+```
+menuentry 'Linux with Secure Launch 6.11-rc7 v11' --unrestricted {
+    insmod part_gpt
+    search --no-floppy --fs-uuid --set=root BOOT-FSUUID
+    slaunch
+    slaunch_module /dce-for-a-given-platform
+    linux /vmlinuz-6.11-rc7-v11-tb root=PARTUUID=ROOT-PARTUUID ro console=ttyS0,115200n8 console=tty0
+
+    # uncomment if used
+    # initrd /initrd-6.11-rc7-v11-tb.img
+}
+```
+
+Things that must be replaced (add additional kernel parameters and GRUB
```

Comment:
`dce-for-a-given-platform` should also be added to this list.

Reply:
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/924f8ee61f342ac95d46d1796f2eb6cd6641726f..eb02c0ec4b2b37327f4ab540d3e3821efabc96d5#diff-9c4b0ba59b4b4f1482abbfcafdbc0e3c091c47cbda592599f4bd03990ee970d3R157-R175